hoodie-client/src/main/java/com/uber/hoodie/io/compact/strategy/PartitionAwareCompactionStrategy.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package com.uber.hoodie.io.compact.strategy;
+
+import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.io.compact.CompactionOperation;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+/**
+ * This strategy orders compactions in reverse order of creation of Hive Partitions.
+ * It helps to compact data in latest partitions first and then older capped at the Total_IO allowed.
+ *
+ */
+public class PartitionAwareCompactionStrategy extends BoundedIOCompactionStrategy {
+    private Comparator<CompactionOperation> comparator;
+
+    public PartitionAwareCompactionStrategy() {
+        super();
+        // For now, use SimpleDateFormat as default partition format
+        String datePartitionFormat = "yyyy/MM/dd";
+        // Sort orders compaction in LastInFirstCompacted order
+        comparator = (CompactionOperation leftC, CompactionOperation rightC) -> {
+            try {
+                Date left = new SimpleDateFormat(datePartitionFormat, Locale.ENGLISH)
+                        .parse(leftC.getPartitionPath());
+                Date right = new SimpleDateFormat(datePartitionFormat, Locale.ENGLISH)
+                        .parse(rightC.getPartitionPath());
+                return left.after(right) ? -1 : right.after(left) ? 1 : 0;
+            } catch(ParseException e) {
+                throw new HoodieException("Invalid Partition Date Format", e);
+            }
+        };
+    }
+
+    /**
+     * Ability to override with a custom comparator, useful when Partition formats are different
+     * @param customComparator
+     */
+    public PartitionAwareCompactionStrategy(Comparator<CompactionOperation> customComparator) {
```

Comment:
is n't the compaction strategy class name passed via config and built using default constructor? 

```
public CompactionStrategy getCompactionStrategy() {
    return ReflectionUtils
        .loadClass(props.getProperty(HoodieCompactionConfig.COMPACTION_STRATEGY_PROP));
  }
```

I don't think this will be customizable..

Reply:
Yeah, didn't notice this. Also, since compaction can be offline, no context from the client can be considered. One possible option I see : 

1. We expose separate configs for compaction strategy sorting format in HoodieCompactionConfig, for example : 1) config for format of partition 2) some config which gives options for sorting like date-sorting, lexicographic sorting. WDYT ?